doc/architecture/layers.sql

```diff
@@ -0,0 +1,112 @@
+-- User table
+CREATE TABLE users (
+    ...
+);
+
+-- Project table
+CREATE TABLE map (
+    ...
+);
+
+-- Custom geometry type
+CREATE DOMAIN geom_type AS GEOMETRY
+CHECK (
+    GeometryType(VALUE) IN ('POINT', 'LINESTRING', 'POLYGON') AND
+    ST_SRID(VALUE) = 4326
+);
+
+/*
+    Still up for discussion whether we want to also use cartesian coordinates
+
+    CREATE DOMAIN geom_cartesian_type AS GEOMETRY
+    CHECK (
+        GeometryType(VALUE) IN ('POINT', 'LINESTRING', 'POLYGON') AND
+        ST_SRID(VALUE) = 3857
+    );
+ */
+
+-- Base Layer table
+CREATE TABLE base_layers (
+    id SERIAL PRIMARY KEY,
+    map_id INTEGER REFERENCES map (id),
+    geometry geom_type,
+    base_type TEXT NOT NULL
+);
+
+-- Fertilize Layer table
+CREATE TABLE fertilize_layers (
+    id SERIAL PRIMARY KEY,
+    map_id INTEGER REFERENCES map (id),
```

Comment:
Is there a particular reason why layers reference maps and not the other way around?
I think doing it this way would make it harder to enforce that maps can only have one layer of a single type.

Reply:
> Is there a particular reason why layers reference maps and not the other way around? 

Yes! I want to decouple the maps from the layers. Maybe it helps to clarify that the layers store the entities that were created on the fertilize layer. The map_id just references to which map it belongs. 

> I think doing it this way would make it harder to enforce that maps can only have one layer of a certain type.

Why do you think that?